engine/src/main/java/com/datatorrent/stram/security/StramWSFilterInitializer.java

```diff
@@ -100,51 +97,20 @@ public String getProxyHostAndPort(Configuration conf)
     Replace with methods from Hadoop when HA support is available
     HttpConfig is not used as it's audience is private as well and it's interface has changed from Hadoop 2.2 to 2.6
   */
-  public String getResolvedRMWebAppURLWithoutScheme(Configuration conf, String rmId) {
-    boolean sslEnabled = conf.getBoolean(
-            CommonConfigurationKeysPublic.HADOOP_SSL_ENABLED_KEY,
-            CommonConfigurationKeysPublic.HADOOP_SSL_ENABLED_DEFAULT);
-    return getResolvedRMWebAppURLWithoutScheme(conf, sslEnabled, (rmId != null) ? "." + rmId : "");
+  public String getResolvedRMWebAppURLWithoutScheme(Configuration conf, String rmId)
+  {
+    InetSocketAddress socketAddress = StramClientUtils.getRMWebAddress(conf, rmId);
+    return socketAddress.getHostString() + ":" + socketAddress.getPort();
   }
 
   /*
     From org.apache.hadoop.yarn.webapp.util.WebAppUtils
     Modified for HA support
   */
-  public String getResolvedRMWebAppURLWithoutScheme(Configuration conf, boolean sslEnabled, String rmPrpKey)
+  public String getResolvedRMWebAppURLWithoutScheme(Configuration conf, boolean sslEnabled, String rmId)
```

Comment:
I actually think it should change because getResolvedRMWebAppURLWithoutScheme was originally an overloaded method with different behaviors among the two different signatures.  They should be consistent especially since they both are public methods